src/main/scala/org/broadinstitute/dsde/firecloud/webservice/UserApiService.scala

```diff
@@ -218,8 +218,10 @@ trait UserApiService extends HttpService with PerRequestCreator with FireCloudRe
           passthrough(UserApiService.samRegisterUserURL, HttpMethods.GET)
         }
       } ~
-      path("userinfo") { requestContext =>
-        requestContext.complete(HttpGoogleServicesDAO.getUserProfile(requestContext))
+      path("userinfo") {
+        requireUserInfo() { userInfo => requestContext =>
+          requestContext.complete(HttpGoogleServicesDAO.getUserProfile(userInfo))
+        }
```

Comment:
this change required by `getUserProfile`'s signature change. It introduces a subtle codepath change. Previously, if you called this endpoint without a token, we'd pass the request through to Google. Google would reject it with a 401 and we'd pass the 401 back up to the user. Now, if you don't supply a token, `requireUserInfo` rejects you and we never talk to Google.

Reply:
+1